### ISSUE TITLE
Training an algorithm - generic

### ISSUE THREAD

**Issue by martinapugliese** [martinapugliese/tales-science-data#147]
There should be a generic intro to ML and how to train an algorithm, with a reference to gradient descent which is in the maths folder and not super visible there.

**Comment by martinapugliese** [comment]
Add a section on cost functions and refer to the gradient descent indeed. For instance the chapter on backpropagation in neural networks should also point there.

**Comment by martinapugliese** [comment]
There is now a subsection for general ML, add there

**Comment by martinapugliese** [comment]
Add training/testing on the same data and in-sample/out-of-sample errors
